File: design-patterns/JavaDesignPatterns/src/com/bigboxcode/designpattern/bridge/uielement/Button.java
package com.bigboxcode.designpattern.bridge.uielement;

public class Button extends UIElement{

    public Button(ColorSchema colorSchema) {
        super(colorSchema);
    }

    @Override
    public void printElement() {
        // Write code for printing element
        System.out.println("Printing Button");

        // Set color schema
        colorSchema.setColor();
    }
}

File: design-patterns/ts/bridge/ui-element/button.ts
// button.ts

import Color from "./color";
import UIElement from "./ui-element";

class Button extends UIElement {
    constructor(color: Color) {
        super(color);
    }

    public printElement(): void {
        this.color.setColor();

        console.log("Printing Button");
    }
}

export default Button;